### PR TITLE
fix: search NFT tab empty on preview

### LINKED ISSUE #7453: search NFT tab empty on preview
image

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Bugfix

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #7453
  - [ ] Requires deployment <snek/rubick/worker>

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [x] My fix has changed UI

image

  ## Copilot Summary
  
### <samp>🤖 Generated by Copilot at b91fabf</samp>

This pull request improves the search suggestion component and the metadata caching strategy. It fixes a bug with reactive data in `SearchSuggestion.vue` and refactors the `processMetadata` function in `cachingStrategy.ts` to use `async/await`.

  
### <samp>🤖 Generated by Copilot at b91fabf</samp>

> _Reactive arrays_
> _Simplify template logic_
> _No more `.value`_

### CODE REVIEW

File: components/search/SearchSuggestion.vue [hunk 1]
```diff
@@ -572,10 +572,10 @@ const updateNftSuggestion = async () => {
     })
     const { nFTEntities } = data.value
     const nftList = unwrapSafe(
-      nFTEntities.value.slice(0, searchSuggestionEachTypeMaxNum)
+      nFTEntities.slice(0, searchSuggestionEachTypeMaxNum)
     )
     const metadataList: string[] = nftList.map(mapNFTorCollectionMetadata)
-    const result: NFTWithMeta[] = []
+    const result: NFTWithMeta[] = reactive([])
```
Comment (roiLeo) [inline components/search/SearchSuggestion.vue:578]:
```suggestion
    const result: NFTWithMeta[] = ref([])
```
isn't it better?

File: utils/cachingStrategy.ts [hunk 2]
```diff
@@ -34,9 +34,8 @@ export const processMetadata = <T>(
 ): void => {
   const metadata = metadataList.map((meta) => meta || '')
 
-  metadata.forEach((m, i) => {
-    const meta = cacheOrFetchMetadata(metadataCache[m], m)
-
+  metadata.forEach(async (m, i) => {
+    const meta = await cacheOrFetchMetadata(metadataCache[m], m)
```
Comment (roiLeo) [inline utils/cachingStrategy.ts:38]: I've already had a problem here (series table if I record), and I managed to do without.
are you sure there won't be side issue with this change?

**Review by roiLeo — APPROVED** [review]
otherwise lg

File: components/search/SearchSuggestion.vue (hunk 1 above)
Comment (Jarsen136) [reply components/search/SearchSuggestion.vue]: Yes, it's better. Thanks

File: utils/cachingStrategy.ts (hunk 2 above)
Comment (Jarsen136) [reply utils/cachingStrategy.ts]: I'm sure there won't be a side issue. I have also updated this line to unify the call of the `processSingleMetadata` function.

**Review by daiagi — APPROVED** [review]

**Pull request merged**